### PR TITLE
stats: Remove remaining calls to Stats::Scope::counter() et al in source/..., found by adding more patterns to the check-format script

### PR DESCRIPTION
**Pull request by jmarantz**
Description: As another step toward removing the string version of counter(), I found a bunch of other calls that were previously missed due to the pattern being scanned for in check_format.
Risk Level: medium
Testing: //test/...
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: include/envoy/grpc/context.h [hunk 1]
```diff
@@ -78,6 +80,8 @@ class Context {
    */
   virtual void chargeResponseMessageStat(const Upstream::ClusterInfo& cluster,
                                          const RequestNames& request_names, uint64_t amount) PURE;
+
+  virtual GoogleGrpcStatNames& googleGrpcStatNames() PURE;
```
Comment (jmarantz) [inline include/envoy/grpc/context.h]: i'll add a doxygen comment here...just want to see if there are other comments first before pushing another change that kicks off CI.

File: .bazelversion [hunk 2]
```diff
@@ -1 +1 @@
-2.0.0
+2.1.1
```
Comment (snowp) [inline .bazelversion]: is this intentional?

File: source/common/grpc/google_grpc_stat_names.h [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+#include "envoy/grpc/status.h"
+
+#include "common/stats/symbol_table_impl.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+namespace Grpc {
+
+struct GoogleGrpcStatNames {
```
[the remark this answers is not in the document]
Comment (jmarantz) [reply source/common/grpc/google_grpc_stat_names.h]: Actually I just went with `StatNames` as we might put anything related to Grpc in here, and Grpc is already in the namespace.

WDYT?

File: .bazelversion (hunk 2 above)
Comment (jmarantz) [reply .bazelversion]: no; removed. I just now adjusted my dev flow so hopefully I don't commit this again.

File: source/common/grpc/google_grpc_stat_names.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (snowp) [reply source/common/grpc/google_grpc_stat_names.h]: yeah that works for me

File: source/common/grpc/stat_names.h [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * Captures symbolized representation for tokens used in grpc stats. These are
+ * broken out so they can be allocated early and used across all gRPC-related
+ * filters.
+ */
+struct StatNames {
+  StatNames(Stats::SymbolTable& symbol_table);
```
Comment (snowp) [inline source/common/grpc/stat_names.h]: `explicit`

File: source/common/grpc/stat_names.h [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Envoy {
+namespace Grpc {
+
+/**
+ * Captures symbolized representation for tokens used in grpc stats. These are
+ * broken out so they can be allocated early and used across all gRPC-related
+ * filters.
+ */
+struct StatNames {
```
Comment (snowp) [inline source/common/grpc/stat_names.h]: do you think it would make sense to provide a base class for these stat name helpers that just has the ctor + the pool field? not sure if it's worth to try to DRY it up given how little code it already is (and it can certainly be addressed in another PR)

Comment (jmarantz) [reply source/common/grpc/stat_names.h]: I don't think it would save anything. You'd have to have a constructor for the derived class to pass the SymbolTable to the base-class constructor; you might well just init the member var.

I guess it saves the member-var declaration but replaces it with an inheritance declaration so it might be one less line of code.
